cmod/conditions.py

```diff
@@ -0,0 +1,226 @@
+      if not(len(conditions["FOV_to_gantry_coordinates"]["diff"]) == 3):
+        raise KeyError("[\"FOV_to_gantry_coordinates\"][\"diff\"] not of length 3")
+      if not(len(conditions["lumi_vs_FOV_center"]["diff"]) == 3):
+        raise KeyError("[\"lumi_vs_FOV_center\"][\"diff\"] not of length 3")
+      if not(len(conditions["lumi_vs_FOV_center"]["FOV_center"]) == 3):
+        raise KeyError("[\"lumi_vs_FOV_center\"][\"FOV_center\"] not of length 3")
+      if not(len(conditions["lumi_vs_FOV_center"]["lumi_center"]) == 3):
+        raise KeyError("[\"lumi_vs_FOV_center\"][\"lumi_center\"] not of length 3")
+      
+      self.gantry_conditions = new_conditions
+      self.increment_use_count()
+
+      return True
+    except KeyError as e:
+      self.logger.error(e)
+      self.logger.error("""
+        The gantry conditions file does not contain the required gantry conditions:
+        'FOV_to_gantry_coordinates', and 'lumi_vs_FOV_center'. Please check the
+        file and the required format and try again.""")
+      # TODO: might want to add logic that if the required conditions are provided run the process to calculate those that are missing or even all of them as don't want to trust an "incomplete" set of conditions
+      return False
+
+  # saves gantry conditions to a file
+  def save_gantry_conditions(self):
+    if self.gantry_conditions_filename is None:
+      self.create_gantry_conditions_filename()
+
+    with open(self.gantry_conditions_filename, 'w') as f:
+      f.write(json.dumps(self.gantry_conditions, indent=2))
+  
+  # returns the gantry conditions
+  def get_gantry_conditions(self):
+    return self.gantry_conditions
+
+  def calculate_gantry_conditions(self, cmd):
+    h_list = []
+    avg_h = 0
+
+    for id in range(1, len(cmd.board.detectors)+1):
+        # vishscan
+        # TODO: confirm inputs to all the  commands
+        cmd_str = """
+        visualhscan --detid={detid} -z {zval} --range 25 --distance 1 
+                    -f=/dev/null
+        """.format(detid=id, zval=5)
+        command = cmd.precmd(cmd_str)
+        sig = cmd.onecmd(command)
+        status = cmd.postcmd(sig, command)
+
+        self.gantry_conditions["fov_to_gantry_coordinates"] = cmd.board.get_detector(id).getVisM(id, 5)
+
+        # vis_center_det vs halign
+        self.calculate_sipm_vis_coordinates(cmd, id, False)
+
+        self.calculate_sipm_lumi_coordinates(cmd, id, False)
+
+    # check if we have multiple H values out of tolerance with each other,
+    # if check_conditions(h_list):
+    #   avg_h = sum(h_list)/len(h_list)
+    #   self.gantry_conditions["lumi_vs_FOV_center"]["diff"] = avg_h
+    # TODO: add the else: an error should be raised such that the operator knows that something is wrong (maybe the gantry head dislodged or was tugged
+
+    self.save_gantry_conditions()
+
+  
+  def calculate_sipm_vis_coordinates(self, cmd, detid=None, save_gantry_conditions_changes=True):
```

Comment:
I don't think we should have classes attempt to call commands, as particularly for the CLI, all commands should be explicitly called by the user. We should move the cooridnates update to calculation post process of the various calibration commands: 

- `halign`: https://github.com/UMDCMS/SiPMCalibControl/blob/master/ctlcmd/motioncmd.py#L266-L287
- `visualhscan`: https://github.com/UMDCMS/SiPMCalibControl/blob/master/ctlcmd/viscmd.py#L176
- `visualcenterdet`: https://github.com/UMDCMS/SiPMCalibControl/blob/master/ctlcmd/viscmd.py#L295

Reply:
@yimuchen can you resolve this conversation, if applicable? If not, let me know further changes that need to be made. Thank you.